Core/include/Acts/Seeding/GbtsDataStorage.hpp

```diff
@@ -9,279 +9,123 @@
 #pragma once
 
 // TODO: update to C++17 style
-#include "Acts/Definitions/Algebra.hpp"
 #include "Acts/Seeding/GbtsGeometry.hpp"
 
-#include <algorithm>
-#include <iostream>
-#include <map>
-#include <numbers>
+#include <array>
+#include <limits>
 #include <vector>
-
 namespace Acts::Experimental {
 
-constexpr std::size_t MAX_SEG_PER_NODE = 1000;  // was 30
-constexpr std::size_t N_SEG_CONNS = 6;          // was 6
-
-// new sp struct
-template <typename space_point_t>
-struct GbtsSP {
-  const space_point_t *SP;  // want inside to have pointer
-  int gbtsID;
-  int combined_ID;
-  float m_phi;
-  float m_r;
-  float m_ClusterWidth;
-  GbtsSP(const space_point_t *sp, int id, int combined_id, float ClusterWidth)
-      : SP(sp),
-        gbtsID(id),
-        combined_ID{combined_id},
-        m_ClusterWidth(ClusterWidth) {
-    m_phi = std::atan2(SP->y(), SP->x());
-    m_r = std::sqrt((SP->x() * SP->x()) + (SP->y() * SP->y()));
-  };
-  float phi() const { return m_phi; }
-  float r() const { return m_r; }
-  float ClusterWidth() const { return m_ClusterWidth; }
-};
-
-template <typename space_point_t>
-class GbtsNode {
- public:
-  explicit GbtsNode(const GbtsSP<space_point_t> &spGbts, float minT = -100.0,
-                    float maxT = 100.0)
-      : m_spGbts(spGbts), m_minCutOnTau(minT), m_maxCutOnTau(maxT) {}
+#define MAX_SEG_PER_NODE 1000  // was 30
+#define N_SEG_CONNS 6          // was 6
 
-  inline void addIn(int i) {
-    if (m_in.size() < MAX_SEG_PER_NODE) {
-      m_in.push_back(i);
-    }
-  }
+class GbtsGeometry;
 
-  inline void addOut(int i) {
-    if (m_out.size() < MAX_SEG_PER_NODE) {
-      m_out.push_back(i);
-    }
-  }
+struct GbtsNode {
```

Comment:
potentially this should rather be  a `class` with writable accessors to the properties. right now it is sort for a mixture of `class` and `struct`